app/containers/SignUp/KeycatSignUpForm.jsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import { compose } from 'redux';
+import { connect } from 'react-redux';
+import { Field, reduxForm } from 'redux-form/immutable';
+import { FormattedMessage } from 'react-intl';
+import { translationMessages } from 'i18n';
+import PropTypes from 'prop-types';
+
+import {
+  required,
+  strLength3x20,
+  validateTelosName,
+} from 'components/FormFields/validate';
+import TextInputField from 'components/FormFields/TextInputField';
+import Button from 'components/Button/Contained/InfoLarge';
+import { SignUpOptions, P } from 'components/SignUpWrapper/SignUpOptions';
+import Checkbox from 'components/Input/Checkbox';
+import IAcceptTerms from 'components/IAcceptTerms';
+
+import SignUp from './index';
+
+import {
+  KEYCAT_ACCOUNT_FIELD,
+  DISPLAY_NAME_FIELD,
+  I_ACCEPT_PRIVACY_POLICY_FIELD,
+} from './constants';
+
+import messages from './messages';
+
+import { Form } from './EmailVerificationForm';
+import loginMessages from '../Login/messages';
+import { REFERRAL_CODE } from '../Login/constants';
+import { getCookie } from '../../utils/cookie';
+import { REFERRAL_CODE_URI } from '../App/constants';
+
+const KeycatSignUpForm = ({ handleSubmit, keycatAccountValue, change }) => (
```

Comment:
Most of this form can be moved into a component that can be reused for signup with Scatter and Sign with Keycat.

Reply:
fixed